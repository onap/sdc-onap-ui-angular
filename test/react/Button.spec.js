import React from 'react';
import Button from '../../src/react/Button.js';

import renderer from 'react-test-renderer';

describe('Button', () => {
	test('Button - Default - Primary', () => {
		const button = renderer.create(<Button>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Default - Primary - Disabled', () => {
		const button = renderer.create(<Button disabled>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Default - White', () => {
		const button = renderer.create(<Button color='white'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Default - Gray', () => {
		const button = renderer.create(<Button color='gray'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Default - Positive', () => {
		const button = renderer.create(<Button color='positive'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Default - Negative', () => {
		const button = renderer.create(<Button color='negative'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Default - Warning', () => {
		const button = renderer.create(<Button color='warning'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Outline - Primary', () => {
		const button = renderer.create(<Button btnType='outline'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Outline - Gray', () => {
		const button = renderer.create(<Button btnType='outline' color='gray'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Outline - Positive', () => {
		const button = renderer.create(<Button btnType='outline' color='positive'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Outline - Negative', () => {
		const button = renderer.create(<Button btnType='outline' color='negative'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Link - Primary', () => {
		const button = renderer.create(<Button btnType='link' color='primary'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Link - Primary - Disabled', () => {
		const button = renderer.create(<Button btnType='link' color='primary' disabled>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Button - Link - Primary - With Icon', () => {
		const button = renderer.create(<Button btnType='link' color='primary' iconName='plus'>Click Me</Button>).toJSON();
		expect(button).toMatchSnapshot();
	});

});
