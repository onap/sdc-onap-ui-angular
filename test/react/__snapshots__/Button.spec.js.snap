// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button - Default - Gray 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  color="gray"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Default - Negative 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  color="negative"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Default - Positive 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  color="positive"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Default - Primary - Disabled 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  disabled={true}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Default - Primary 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Default - Warning 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  color="warning"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Default - White 1`] = `
<button
  className="sdc-button sdc-button__primary   "
  color="white"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Link - Primary - Disabled 1`] = `
<button
  className="sdc-button sdc-button__link   "
  color="primary"
  disabled={true}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Link - Primary - With Icon 1`] = `
<button
  className="sdc-button sdc-button__link   plus"
  color="primary"
  disabled={false}
  onClick={undefined}
>
  <div
    className="svg-icon-wrapper     right"
    disabled={undefined}
    onClick={undefined}
  >
    <svg />
    <span
      className="svg-icon-label "
    >
      Click Me
    </span>
  </div>
</button>
`;

exports[`Button Button - Link - Primary 1`] = `
<button
  className="sdc-button sdc-button__link   "
  color="primary"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Outline - Gray 1`] = `
<button
  className="sdc-button sdc-button__outline   "
  color="gray"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Outline - Negative 1`] = `
<button
  className="sdc-button sdc-button__outline   "
  color="negative"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Outline - Positive 1`] = `
<button
  className="sdc-button sdc-button__outline   "
  color="positive"
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;

exports[`Button Button - Outline - Primary 1`] = `
<button
  className="sdc-button sdc-button__outline   "
  disabled={false}
  onClick={undefined}
>
  Click Me
</button>
`;
